whelktool/scripts/cleanups/2020/11/lxl-3340-add-idkbse-uris.groovy

```diff
@@ -0,0 +1,30 @@
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+PrintWriter failedUpdating = getReportWriter("failed-updates")
+
+String where = """
+        collection = 'auth' AND (
+        data#>>'{@graph,1,@type}' = 'Topic' OR
+        data#>>'{@graph,1,@type}' = 'ComplexSubject' OR
+        data#>>'{@graph,1,@type}' = 'Temporal' OR
+        data#>>'{@graph,1,@type}' = 'Geographic' OR
+        data#>>'{@graph,1,@type}' = 'TopicSubdivision')
+        AND (
+        data#>>'{@graph,1,inScheme,@id}' = 'https://id.kb.se/term/barn' OR
+        data#>>'{@graph,1,inScheme,@id}' = 'https://id.kb.se/term/sao')
+        AND NOT (
+        data#>>'{@graph,0,mainEntity}' LIKE '%id.kb.se%')
+        """
+
+selectBySqlWhere(where) { data ->
+    String librisUri = data.graph[1]['@id']
+    String idkbseUri = data.graph[1].inScheme['@id'] + '/' + data.graph[1].prefLabel
+
+    data.graph[1]['@id'] = idkbseUri
+    data.graph[1].sameAs += ['@id':librisUri]
```

Comment:
Neat! I didn't know the += operator worked on lists! Thanks for that ;)

Reply:
Me neither! I used it unintentionally first (kind of copy pasted your pseudo-code) and then realized it actually worked. So thank you :)